### PR TITLE
Hermes retrying mechanism for packet clearing

### PR DESCRIPTION
**Pull request by adizere**
Part of the work towards closing #1124 

## Description

Changes in this PR:
- enables packet clearing to be retried every `X` blocks, parametrized in config.toml
- makes the `rpc_timeout` from config file reflect in the timeout for "waiting on block commits" 
- adds an explanation to the `TxNoConfirmation` error message

______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Review by adizere — COMMENTED** [review]
LGTM!

File: relayer-cli/src/commands/tx/packet.rs [hunk 1]
```diff
@@ -22,6 +24,26 @@ pub struct TxRawPacketRecvCmd {
...
+        let src_chain_config = config.find_chain_mut(&self.src_chain_id).ok_or_else(|| {
+            FrameworkErrorKind::ComponentError.context("missing src chain configuration")
+        })?;
+
+        if let Some(ref key_name) = self.key {
+            src_chain_config.key_name = key_name.to_string();
+        }
+
+        Ok(config)
+    }
```
Comment (ancazamfir) [inline relayer-cli/src/commands/tx/packet.rs:46]: maybe we undo these changes? i only used this for testing and we can always use a different config for the CLI

File: relayer/src/link.rs [hunk 2]
```diff
@@ -412,7 +412,7 @@ impl RelayPath {
 
     /// Should not run more than once per execution.
     pub fn clear_packets(&mut self, above_height: Height) -> Result<(), LinkError> {
-        if self.clear_packets {
+        if self.clear_packets || above_height.revision_height % 100 == 0 {
```
Comment (ancazamfir) [inline relayer/src/link.rs:415]: `100` should be configurable

File: relayer-cli/src/commands/tx/packet.rs (hunk 1 above)
Comment (adizere) [reply relayer-cli/src/commands/tx/packet.rs]: Good idea!

**Review by romac — APPROVED** [review]

File: relayer/src/supervisor/spawn.bak.rs [hunk 3]
```diff
@@ -0,0 +1,550 @@
+use anomaly::BoxError;
```
Comment (adizere) [inline relayer/src/supervisor/spawn.bak.rs:1]: I guess this is from another parallel work. Safe to delete?
